synapse/config/logger.py

```diff
@@ -201,3 +170,55 @@ def _log(event):
     )
     if not config.no_redirect_stdio:
         print("Redirected stdout/stderr to logs")
+
+
+def _reload_stdlib_logging(*args, log_config=None):
+    logger = logging.getLogger("")
+
+    if not log_config:
+        logger.warn("Reloaded a blank config?")
+
+    logging.config.dictConfig(log_config)
+
+
+def setup_logging(config, use_worker_options=False):
+    """
+    Set up the logging subsystem.
+
+    Args:
+        config (LoggingConfig | synapse.config.workers.WorkerConfig):
+            configuration data
+
+        use_worker_options (bool): True to use the 'worker_log_config' option
+            instead of 'log_config'.
+
+        register_sighup (func | None): Function to call to register a
+            sighup handler.
+    """
+    log_config = config.worker_log_config if use_worker_options else config.log_config
+
+    def read_config(*args, callback=None):
+        if log_config is None:
+            return None
+
+        log_config_body = yaml.safe_load(FilePath(log_config).getContent())
+        if args:
+            logging.info("Reloaded log config from %s due to SIGHUP", log_config)
+        if callback:
+            callback(log_config=log_config_body)
+        return log_config_body
+
+    log_config_body = read_config()
+
+    if log_config_body and log_config_body.get("version") == 2:
```

Comment:
is version 2 format defined somewhere? https://docs.python.org/3.7/library/logging.config.html#dictionary-schema-details suggests that anything other than 1 is reserved.

Reply:
It's not, but out of having an entirely new format and repurposing the stdlib style that will likely never change, the latter seemed to have less friction.

Reply:
I'm not entirely convinced, but I can't quite put my finger on why, so I guess it's ok.